libsolidity/experimental/codegen/IRGeneratorForStatements.h

```diff
@@ -34,6 +38,62 @@ class IRGeneratorForStatements: public ASTConstVisitor
 	IRGeneratorForStatements(IRGenerationContext& _context): m_context(_context) {}
 
 	std::string generate(ASTNode const& _node);
+
+	static std::size_t stackSize(IRGenerationContext const& _context, Type _type)
+	{
+		TypeSystemHelpers helper{_context.analysis.typeSystem()};
+		_type = _context.env->resolve(_type);
+		solAssert(std::holds_alternative<TypeConstant>(_type), "No monomorphized type.");
+
+		// type -> # stack slots
+		// unit, itself -> 0
+		// void, literals(integer), typeFunction -> error (maybe generate a revert)
+		// word, bool, function -> 1
+		// pair -> sum(stackSize(args))
+		// user-defined -> stackSize(underlying type)
+		TypeConstant typeConstant = std::get<TypeConstant>(_type);
+		if (helper.isPrimitiveType(_type, PrimitiveType::Unit) ||
+			helper.isPrimitiveType(_type, PrimitiveType::Itself))
+			return 0;
+		else if (helper.isPrimitiveType(_type, PrimitiveType::Bool) || helper.isPrimitiveType(_type, PrimitiveType::Word))
+		{
+			solAssert(typeConstant.arguments.empty(), "Primitive type Bool or Word should have no arguments.");
+			return 1;
+		}
+		else if (helper.isFunctionType(_type))
+			return 1;
+		else if (
+			helper.isPrimitiveType(_type, PrimitiveType::Integer) ||
+			helper.isPrimitiveType(_type, PrimitiveType::Void) ||
+			helper.isPrimitiveType(_type, PrimitiveType::TypeFunction))
+			solThrow(langutil::CompilerError, "Primitive type Void, Unit or TypeFunction do not have stack slot"); // FIXME: proper error
```

Comment:
```suggestion
			solAssert(false, "Attempted to query the stack size of a type without stack representation.");
```
We can keep this as an assertion - we should catch everything that may result in this in analysis already eventually.